### PR TITLE
[MIG] jasper_reports: Migrated Jasper Report module for Odoo 14

### PR DESCRIPTION
**Pull request by junusd**
Hi,

This is migrated Jasper Report module for Odoo 14.

Regards,

### CODE REVIEW

**Comment by JayVora-SerpentCS** [comment]
@hiren-serpentcs review each segment and update.
Thank you @junusd

**Comment by holybank** [comment]
The company I'm working with is highly interested in this module for Odoo version 14.
It is very likely that if this migration is finished then the company will purchase the module.
So, if you can make it happen sooner, it would be great.
Thank you.

**Comment by JayVora-SerpentCS** [comment]
@holybank the module is opensource.
However, if your firm wants to expedite the efforts for you only, we can allocate a team for you.
Kindly contact on contact@serpentcs.com
Thanks,

File: jasper_reports/controllers/main.py [hunk 1]
```diff
@@ -69,12 +71,67 @@ def report_routes(self, reportname, docids=None, converter=None, **data):
...
+                else:
+                    # Particular report:
+                    data = dict(url_decode(url.split('?')[1]).items())  # decoding the args represented in JSON
+                    if 'context' in data:
+                        context, data_context = json.loads(context or '{}'), json.loads(data.pop('context'))
+                        context = json.dumps({**context, **data_context})
+                    response = self.report_routes(reportname, converter=converter, context=context, **data)
+
+                report = request.env['ir.actions.report']._get_report_from_name(reportname)
+                filename = "%s.%s" % (report.name, extension)
```
Comment (hiren-serpentcs) [inline jasper_reports/controllers/main.py:117]:
```suggestion
                extension = report.jasper_output or "pdf"
                filename = "%s.%s" % (report.name, extension)
```

**Comment by hiren-serpentcs** [comment]
@JayVora-SerpentCS 
LGTM :+1:

**Review by hiren-serpentcs — APPROVED** [review]

**Pull request merged**